### PR TITLE
Remove deprecated curses (cli) rendering backend.

### PR DESCRIPTION
**Pull request by supakeen**
Since this removes a big chunk of code I'd thought everyone should have a look at it, @sebastinas, @thomasballinger.

### CODE REVIEW

File: bpython/translations/bpython.pot [hunk 1]
```diff
@@ -61,42 +61,14 @@ msgstr ""
 msgid "File to execute and additional arguments passed on to the executed script."
 msgstr ""
 
-#: bpython/cli.py:320 bpython/curtsiesfrontend/interaction.py:107
```
Comment (sebastinas) [inline bpython/translations/bpython.pot:64]: The `#: 'bpython/curtsiesfrontend/interaction.py:107` should stay.

File: doc/sphinx/source/tips.rst [hunk 2]
```diff
@@ -16,7 +16,7 @@ equivalent file.
 
 .. code-block:: bash
 
-  alias bpython3.5='PYTHONPATH=~/python/bpython python3.5 -m bpython.cli'
+  alias bpython3.5='PYTHONPATH=~/python/bpython python3.5 -m bpython.curtsies'
```
Comment (sebastinas) [inline doc/sphinx/source/tips.rst:19]: Let's just use `… -m bpython` here.

File: bpython/translations/bpython.pot (hunk 1 above)
Comment (supakeen) [reply bpython/translations/bpython.pot:64]: Oops, bit too heavy on the delete.

File: doc/sphinx/source/tips.rst (hunk 2 above)
Comment (supakeen) [reply doc/sphinx/source/tips.rst:19]: I like that more as well.

**Review by sebastinas — APPROVED** [review]

**Comment by thomasballinger** [comment]
hooray!

**Review by thomasballinger — APPROVED** [review]

**Comment by thomasballinger** [comment]
approved to get this out of my queue and because I approve of the idea! but didn't review in detail

**Comment by supakeen** [comment]
You know what, I'm ready to hit merge on this (after solving conflicts). @sebastinas Are you in agreement?

**Comment by sebastinas** [comment]
Let's do it.

**Comment by supakeen** [comment]
Ok, did the rebase, if you re-ack I'll merge @sebastinas :)

**Comment by sebastinas** [comment]
Looks good.

**Pull request merged**